cpp/src/parquet/arrow/reader.cc

```diff
@@ -1134,6 +1138,42 @@ Status FileReaderImpl::DecodeRowGroups(const std::vector<int>& row_groups,
   return (*out)->Validate();
 }
 
+Future<std::shared_ptr<Table>> FileReaderImpl::DecodeRowGroups(
```

Comment:
Is there a way to reconcile this with the sync `DecodeRowGroups` version?

Reply:
I see you finally did it :-) Great, thank you!

Reply:
Sorry, I should probably have pinged you when I updated the PR.